android-lib/src/main/java/com/kinvey/android/async/AsyncBatchPushRequest.java

```diff
@@ -0,0 +1,267 @@
+            }
+        }
+
+        pushResponse.setEntities(resultAllItems);
+        return pushResponse;
+    }
+
+    private List<GenericJson> processSingleSyncItems(KinveyPushBatchResponse pushResponse, List<SyncItem> syncItems) throws IOException {
+        String id;
+        T item = null;
+        SyncRequest syncRequest = null;
+        List<GenericJson> resultSingleItems = new ArrayList<>();
+        if (syncItems != null) {
+            for (SyncItem syncItem : syncItems) {
+                id = syncItem.getEntityID().id;
+                SyncItem.HttpVerb requestMethod = syncItem.getRequestMethod();
+                switch (requestMethod) {
+                    case SAVE: // the SAVE case need for backward compatibility
+                    case POST:
+                    case PUT:
+                        item = cache.get(id);
+                        if (item == null) {
+                            // check that item wasn't deleted before
+                            // if item wasn't found, it means that the item was deleted from the Cache by Delete request and the item will be deleted in case:DELETE
+                            manager.deleteCachedItems(client.query().equals("meta.id", syncItem.getEntityID().id).notEqual(Constants.REQUEST_METHOD, Constants.DELETE));
+                            continue;
+                        }
+                        if (requestMethod != SyncRequest.HttpVerb.POST) {
+                            syncRequest = manager.createSyncRequest(collection, networkManager.saveBlocking(item));
+                        }
+                        break;
+                    case DELETE:
+                        syncRequest = manager.createSyncRequest(collection, networkManager.deleteBlocking(id));
+                        break;
+                }
+                try {
+                    if (SyncRequest.HttpVerb.POST.equals(requestMethod)) {
+                        batchSyncItems.add(syncItem);
+                    } else {
+                        GenericJson resultItem = runSingleSyncRequest(syncRequest);
+                        resultSingleItems.add(resultItem);
+                        pushResponse.setSuccessCount(++progress);
+                        manager.deleteCachedItem((String) syncItem.get(_ID));
+                    }
+                } catch (IOException e) {
+                    KinveyUpdateSingleItemError err = new KinveyUpdateSingleItemError(e, item);
+                    errors.add(err);
+                } catch (Exception e) {
+                    callback.onFailure(e);
+                }
+                callback.onProgress(pushResponse.getSuccessCount(), fullCount);
+            }
+        }
+        return resultSingleItems;
+    }
+
+    private KinveySyncSaveBatchResponse processBatchSyncRequest(List<SyncItem> syncItems, List<T> saveItems) throws IOException {
+        SyncRequest syncRequest = manager.createSaveBatchSyncRequest(collection, networkManager, saveItems);
```

Comment:
I believe you need to split this request into multiple requests if the length/size of saveItems is greater than `100`

Reply:
It will be a separated task for this feature.

Reply:
This work is tracked by https://kinvey.atlassian.net/browse/MLIBZ-3089.